samples/java-eventsourced-counter-pojo/src/main/java/org/example/domain/CounterState.java

```diff
@@ -0,0 +1,29 @@
+package org.example.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class CounterState {
+
+  public final int value;
+
+  @JsonCreator
+  public CounterState(@JsonProperty("value") int value) {
```

Comment:
JsonProperty shouldn't be needed. Maybe you haven't added the `-parameters` Java compiler option?

Reply:
I will try this out.

Reply:
This is removed now. But I couldn't remove the `@JsonCreator` one, only `@JsonProperty`.

Reply:
So, it seems that it's needed when the constructor has only one argument which is the case in all the example here. TIL

Reply:
Yes, single arg constructor requires JsonCreator